datumaro/plugins/imagenet_format.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from collections import defaultdict
+import os
+import os.path as osp
+from glob import glob
+import logging as log
+
+from datumaro.components.extractor import (DatasetItem, Label,
+    LabelCategories, AnnotationType, SourceExtractor, Importer
+)
+from datumaro.util.log_utils import logging_disabled
+from datumaro.components.converter import Converter
+
+
+class ImagenetPath:
+    LABELS_FILE = 'synsets.txt'
+    IMAGES_DIR = 'data'
+
+
+class ImagenetExtractor(SourceExtractor):
+    def __init__(self, path):
+        assert osp.isfile(path), path
+        self._path = path
+        self._dataset_dir = osp.dirname(path)
+
+        super().__init__(subset=osp.splitext(osp.basename(path))[0])
+
+        labels = osp.join(osp.dirname(path), ImagenetPath.LABELS_FILE)
+        if isinstance(labels, str):
+            labels = self._parse_labels(labels)
+        elif isinstance(labels, list):
+            assert all(isinstance(lbl, str) for lbl in labels), labels
+        else:
+            raise TypeError("Unexpected type of 'labels' argument: %s" % labels)
+
+        self._categories = self._load_categories(labels)
+        self._items = list(self._load_items(path).values())
+
+    @staticmethod
+    def _parse_labels(path):
+        with open(path, encoding='utf-8') as labels_file:
+            return [s.strip() for s in labels_file]
+
+    def _load_categories(self, labels):
+        label_cat = LabelCategories()
+        for label in labels:
+            label_cat.add(label)
+
+        return { AnnotationType.label: label_cat }
+
+    def _load_items(self, path):
+        items = defaultdict(dict)
+
+        with open(path, encoding='utf-8') as f:
+            for line in f:
+                image_name, label_id = line.split()
+                items[image_name[:-4]] = DatasetItem(
+                    id=image_name[:-4], subset=self._subset,
+                    image=image_name,
+                    annotations=[Label(label=label_id)])
+        return items
+
+
+class ImagenetImporter(Importer):
+
+    @classmethod
+    def detect(cls, path):
+        with logging_disabled(log.WARN):
+            return (len(cls.find_sources(path)) != 0)
+
+    def __call__(self, path, **extra_params):
```

Comment:
I think, you don't need your own implementation for `detect` an `call` after the simplification patch.